pandas/core/window_/aggregators.py

```diff
@@ -61,19 +57,21 @@ class SubtractableAggregator(BaseAggregator):
     def __init__(self, values: np.ndarray, agg: AggKernel) -> None:
         super().__init__(values)
         self.agg = agg
+        self.previous_start = -1
+        self.previous_end = -1
 
-    def query(
-        self, start: int, stop: int, previous_start: int, previous_end: int
-    ) -> Scalar:
+    def query(self, start: int, stop: int) -> Scalar:
```

Comment:
do we have need for a .reset() method? basically this is what you do in ‘’__init__’’ but is externally visible
I think we might need this for perf 
@cpcloud

Reply:
If the same `SubtractableAggregator` instance and calls `query` twice, a `reset` method would be needed. Based on our spec we'll won't be reusing the same aggregator (I think), but I suppose a user can make their own top level functions with these aggregators?